app/controllers/actions/IdentifierAction.scala

```diff
@@ -43,33 +44,37 @@ class AuthenticatedIdentifierAction @Inject() (
 
     implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequestAndSession(request, request.session)
 
-    authorised().retrieve(Retrievals.internalId) {
-      _.map {
-        internalId => block(IdentifierRequest(request, internalId))
-      }.getOrElse(throw new UnauthorizedException("Unable to retrieve internal Id"))
+    authorised().retrieve(Retrievals.internalId and Retrievals.allEnrolments) {
+      case Some(internalId) ~ enrolments => getSubscriptionId(request, enrolments, internalId, block)
+      case _                             => throw new UnauthorizedException("Unable to retrieve internal Id")
```

Comment:
Instead of UnauthorizedException, we should throw AuthorisationException, So it can be redirected to UnauthorisedController.onPageLoad()  page
```suggestion
      case _                             => throw new AuthorisationException("Unable to retrieve internal Id")
```